aten/src/ATen/native/mkldnn/Matmul.cpp

```diff
@@ -0,0 +1,75 @@
+#include <ATen/ATen.h>
+#include <ATen/Config.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/native/mkldnn/Matmul.h>
+#if !AT_MKLDNN_ENABLED()
+
+namespace at {
+namespace native {
+
+void mkldnn_matmul(
+    const Tensor &mat1,
+    const Tensor &mat2,
+    Tensor &result,
+    float beta,
+    float alpha) {
+  TORCH_CHECK(false, "mkldnn_matmul: ATen not compiled with MKLDNN support");
+}
+} // namespace native
+} // namespace at
+
+#else // AT_MKLDNN_EBABLED
+
+#include <ATen/native/mkldnn/MKLDNNCommon.h>
+#include <ATen/native/mkldnn/Utils.h>
+
+namespace at {
+namespace native {
+
+void mkldnn_matmul(
+    const Tensor &mat1,
+    const Tensor &mat2,
+    const Tensor &result,
+    float beta,
+    float alpha) {
+  TORCH_CHECK(mat1.dim() == 2 || mat1.dim() == 3, "mkldnn_matmul:  expect mat1 to be 2-D or 3-D tensor");
+  TORCH_CHECK(mat2.dim() == 2 || mat2.dim() == 3, "mkldnn_matmul:  expect mat2 to be 2-D or 3-D tensor");
+  TORCH_CHECK(mat1.scalar_type() == at::kBFloat16, "mkldnn_matmul:  only enabled for bf16 path");
+  TORCH_CHECK(mat2.scalar_type() == at::kBFloat16, "mkldnn_matmul:  only enabled for bf16 path");
+  TORCH_CHECK(result.scalar_type() == at::kBFloat16, "mkldnn_matmul:  only enabled for bf16 path");
+  TORCH_CHECK(mkldnn_bf16_device_check(),
+    "mkldnn_matmul: mkldnn_matmul bf16 path needs the cpu support avx512bw, avx512vl and avx512dq");
+  // In aten, expect beta * result = 0 if beta = 0
```

Comment:
you'd better give an explaination of API difference of mkldnn and pytorch matmul, aka. why we need post op here.